lib/stitches/auto_fill.py

```diff
@@ -47,18 +49,42 @@ def is_segment(self):
 
 
 @debug.time
-def auto_fill(shape,
-              angle,
-              row_spacing,
-              end_row_spacing,
-              max_stitch_length,
-              running_stitch_length,
-              running_stitch_tolerance,
-              staggers,
-              skip_last,
-              starting_point,
-              ending_point=None,
-              underpath=True):
+def auto_fill(shape: Shape,
+              angle: float,
+              row_spacing: int,
+              end_row_spacing: int,
+              max_stitch_length: float,
+              running_stitch_length: float,
+              running_stitch_tolerance: float,
+              staggers: int,
+              skip_last: bool,
+              starting_point: PointLike,
+              ending_point: PointLike = None,
+              underpath: bool = True) -> List[Stitch]:
+    """Outputs a stitch pattern that fills a given polygon.
+
+    :param shape: The shape to fill.
+    :param angle: The angle of the grating pattern for the stitches.
+    :param row_spacing: The spacing of the rows
+    :param end_row_spacing: The row spacing at the end of the grating. Used
+        to produce gradient blending. See
+        https://inkstitch.org/docs/features/#color-blending
+    :param max_stitch_length: The maximum stitch length, in millimeters
+    :param running_stitch_length:
+    :param running_stitch_tolerance:
+    :param staggers: How many times a stitch stagger occurs before the stagger
+        pattern repeats. See py:function::stitch_row in
+        :py:mod:`lib.stitches.fill`.
+    :param skip_last: Whether to skip the last stitch in each row. FIXME: why
+        is this useful?
```

Comment:
Why is skipping the last stitch in each row useful?

Reply:
When sewing fill stitch, the rows are often 0.25mm apart or even closer.  Depending on the fill angle, the last stitch in one row and the first stitch in the next can be very close together.  They're effectively redundant, and in fact you can run into problems where the second stitch actually shreds the thread!  Skipping the last stitch avoids that risk and speeds things up, and it's usually visually indistinguishable.

Ink/Stitch currently doesn't do the "skip last stitch" well.  It's a bug I haven't gotten to the bottom of.

Reply:
Added a comment, thanks!